src/scheduler/controller.rs

```diff
@@ -128,7 +143,11 @@ impl<VM: VMBinding> GCController<VM> {
                 CoordinatorMessage::Finish => {}
             }
         }
-        self.scheduler.deactivate_all();
+
+        {
+            let _guard = self.scheduler.worker_monitor.0.lock().unwrap();
```

Comment:
Maybe add some comments about why we need to acquire the lock here (like 'make sure no worker is trying to update buckets or schedule sentinel while we are deactivating buckets').

Reply:
I added a comment.